src/utils/fieldUtils/validateAsync.js

```diff
@@ -2,7 +2,7 @@ export default async function validateAsync({ fieldProps, fields, formProps }) {
   const value = fieldProps.get('value');
   const asyncRule = fieldProps.get('asyncRule');
 
-  if (!asyncRule) return { expected: true };
+  if (!asyncRule || !value) return { expected: true };
```

Comment:
Let's move this condition to https://github.com/kettanaito/react-advanced-form/blob/master/src/Form.jsx#L255:
```js
let shouldValidate = !validatedSync || (expected && !validatedAsync && asyncRule && !value); // ???
```

The idea is that utility methods are really dumb, they should not posses much logic related to the necessity of the validation. That is the responsibility of the Form. Please try it and let me know. Thanks.